### PR TITLE
[DebugBundle] Remove inlined dumps on XHR

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | partially #14608 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 1]
```diff
@@ -135,13 +135,12 @@ public function dump(Data $data)
 
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
-        if ($this->requestStack && $this->requestStack->getMasterRequest() !== $request) {
+        if (($this->requestStack && $this->requestStack->getMasterRequest() !== $request) || $request->isXmlHttpRequest()) {
             return;
         }
 
         // In all conditions that remove the web debug toolbar, dumps are written on the output.
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: this comment is not really accurrate anymore

**Comment by nicolas-grekas** [comment]
I added a check for the `Origin` header, so that programmatic calls (either XHR, embedded asset or CORS requests) don't break with invalid content.
This PR is ready for the first most basic round of tweaking regarding #14608
The next step is giving easy control over dumps destination, but that's for an other PR.

**Comment by jakzal** [comment]
seems :+1:

**Comment by xabbuh** [comment]
:+1:

**Comment by aitboudad** [comment]
:+1:

**Pull request merged**